cirq/experiments/fidelity_estimation.py

```diff
@@ -470,17 +505,42 @@ def __enter__(
     def __exit__(self, exc_type, exc_val, exc_tb):
         pass
 
-    def update(self, increment: int):
+    def update(self, n: int = 1):
         pass
 
 
+@dataclass(frozen=True)
+class _ZippedCircuit:
+    """A fully-wide circuit made by zipping together a bunch of two-qubit circuits
+    and its provenance data.
+
+    Args:
+        wide_circuit: The zipped circuit on all pairs
+        pairs: The pairs of qubits operated on in the wide circuit.
+        layer_i: The index of the GridInteractionLayer to which the pairs correspond.
+        combination_i: The row index of the combinations matrix that identifies this
+            particular combination of component narrow circuits.
+        combination: The row of the combinations matrix. Each entry is an index
+            into the (narrow) `circuits` library. Each entry indexes the
+            narrow circuit operating on the corresponding pair in `pairs`.
+    """
+
+    wide_circuit: 'cirq.Circuit'
+    pairs: List[Tuple['cirq.Qid', 'cirq.Qid']]
+    layer_i: int
```

Comment:
I guess it's fine that layer itself might be None?

Reply:
yes, see longer description above. The `layer` might be `None`. `layer_i` is just a contiguous 0-based index into which spacial configuration is being characterized. Imagine using the calibration API where this is the moment index of the circuit, where each moment typifies a configuration of 2q gates.